components/canvas/raqote_backend.rs

```diff
@@ -716,9 +716,9 @@ impl ToRaqoteStyle for RGBA {
 
     fn to_raqote_style(self) -> Self::Target {
         raqote::SolidSource {
-            r: self.red,
-            g: self.green,
-            b: self.blue,
+            r: (self.red_f32() * self.alpha_f32()).round() as u8,
+            g: (self.green_f32() * self.alpha_f32()).round() as u8,
+            b: (self.blue_f32() * self.alpha_f32()).round() as u8,
```

Comment:
https://doc.servo.org/cssparser/struct.RGBA.html#method.red_f32 says it returns a value between 0 and 1, so I suspect you actually want `(self.red as f32 * self.alpha as f32).round() as u8` instead.

Reply:
You're right, the expression would either return `1` or `0`, which makes absolutely no sense at all. Thanks for pointing this out!

Reply:
Fixing this did not help in this case, because the code is not reached when running that exact test case. I'll keep debugging this.